File: Workshop 3/src/BlackJack/model/Game.java
/*
 * Game.java
 * Date: 26 Oktober, 2014
 * Author: Tobias Ohlsson (edited by Robert Willem Hallink)
 */
package BlackJack.model;

/*
 * Mostly left untouched.
 * Added the public boolean method Stand, in order to finish the code.
 */
public class Game {
    
    private final Dealer m_dealer;
    private final Player m_player;
    
    public Game() {
        
        m_dealer = new Dealer(new BlackJack.model.rules.RulesFactory());
        m_player = new Player();
        
    }
    
    public boolean IsGameOver()             { return m_dealer.IsGameOver(); }
    public boolean IsDealerWinner()         { return m_dealer.IsDealerWinner(m_player); }
    public boolean NewGame()                { return m_dealer.NewGame(m_player); }
    public boolean Hit()                    { return m_dealer.Hit(m_player); }
    public Iterable<Card> GetDealerHand()   { return m_dealer.GetHand(); }
    public Iterable<Card> GetPlayerHand()   { return m_player.GetHand(); }
    public int GetDealerScore()             { return m_dealer.CalcScore(); }
    public int GetPlayerScore()             { return m_player.CalcScore(); }
    public boolean Stand()                  { return m_dealer.Stand(m_dealer); }
    
}

File: Workshop 3/Files/Peer Review/Materials - Eddy Proca/model/Game.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BlackJack.model
{
    class Game
    {
        private model.Dealer m_dealer;
        private model.Player m_player;
        public Game()
        {
            m_dealer = new Dealer(new rules.RulesFactory());
            m_player = new Player();
        }
        public void Register(CardDealtListener a_subscriber)
        {
            m_dealer.Register(a_subscriber);
        }
        public bool IsGameOver()
        {
            return m_dealer.IsGameOver();
        }
        public bool IsDealerWinner()
        {
            return m_dealer.IsDealerWinner(m_player);
        }
        public bool NewGame()
        {
            return m_dealer.NewGame(m_player);
        }
        public bool Hit()
        {
            return m_dealer.Hit(m_player);
        }
        public bool Stand()
        {
            return m_dealer.Stand();
        }
        public IEnumerable<Card> GetDealerHand()
        {
            return m_dealer.GetHand();
        }
        public IEnumerable<Card> GetPlayerHand()
        {
            return m_player.GetHand();
        }
        public int GetDealerScore()
        {
            return m_dealer.CalcScore();
        }
        public int GetPlayerScore()
        {
            return m_player.CalcScore();
        }
    }
}
